src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -0,0 +1,56 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repositories.FacultyRepository;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+@Service
+public class FacultyService {
+    private final FacultyRepository facultyRepository;
+
+    public FacultyService(FacultyRepository facultyRepository) {
+        this.facultyRepository = facultyRepository;
+    }
+
+    public Faculty createFaculty(Faculty faculty) {
+        return facultyRepository.save(faculty);
+    }
+
+    public Faculty readFaculty(Long id) {
+        if (facultyRepository.existsById(id)) return facultyRepository.findById(id).get();
+        return null;
```

Comment:
Это всё делается одной строкой, если правильно использовать Optional:
```suggestion
        return facultyRepository.findById(id).orElse(null);
```